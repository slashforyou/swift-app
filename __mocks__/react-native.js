// Mock complet de React Native pour les tests
const React = require('react');

const View = 'View';
const Text = 'Text';
const TouchableOpacity = 'TouchableOpacity';
const Image = 'Image';
const TextInput = 'TextInput';
const ScrollView = 'ScrollView';
const FlatList = 'FlatList';

// Modal component qui respecte la prop visible
const Modal = ({ children, visible }) => {
  return visible ? children : null;
};

const ActivityIndicator = 'ActivityIndicator';
const SafeAreaView = 'SafeAreaView';
const KeyboardAvoidingView = 'KeyboardAvoidingView';
const Pressable = 'Pressable';
const Button = 'Button';
const Switch = 'Switch';
const RefreshControl = 'RefreshControl';

const Platform = {
  OS: 'ios',
  Version: 14,
  select: (obj) => obj.ios || obj.default,
  isPad: false,
  isTVOS: false,
  isTV: false,
};

const StyleSheet = {
  create: (styles) => styles,
  flatten: (styles) => styles,
  compose: (a, b) => [a, b],
};

const Dimensions = {
  get: jest.fn(() => ({ width: 375, height: 812 })),
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
};

const Alert = {
  alert: jest.fn(),
  prompt: jest.fn(),
};

const Animated = {
  View: 'Animated.View',
  Text: 'Animated.Text',
  Image: 'Animated.Image',
  ScrollView: 'Animated.ScrollView',
  Value: jest.fn(() => ({
    setValue: jest.fn(),
    interpolate: jest.fn(() => ({
      setValue: jest.fn(),
    })),
  })),
  timing: jest.fn(() => ({
    start: jest.fn((callback) => callback && callback()),
    stop: jest.fn(),
    reset: jest.fn(),
  })),
  spring: jest.fn(() => ({
    start: jest.fn((callback) => callback && callback()),
    stop: jest.fn(),
    reset: jest.fn(),
  })),
  decay: jest.fn(() => ({
    start: jest.fn((callback) => callback && callback()),
    stop: jest.fn(),
    reset: jest.fn(),
  })),
  sequence: jest.fn(),
  parallel: jest.fn(),
  stagger: jest.fn(),
  loop: jest.fn(),
  event: jest.fn(),
  createAnimatedComponent: (component) => component,
};

const Keyboard = {
  dismiss: jest.fn(),
  addListener: jest.fn(() => ({ remove: jest.fn() })),
  removeListener: jest.fn(),
  removeAllListeners: jest.fn(),
};

const PixelRatio = {
  get: jest.fn(() => 2),
  getFontScale: jest.fn(() => 1),
  getPixelSizeForLayoutSize: jest.fn((size) => size * 2),
  roundToNearestPixel: jest.fn((size) => size),
};

const NativeModules = {
  DevMenu: {
    reload: jest.fn(),
    show: jest.fn(),
  },
  PlatformConstants: {
    isTesting: true,
    reactNativeVersion: { major: 0, minor: 73, patch: 0 },
  },
};

const NativeEventEmitter = jest.fn(() => ({
  addListener: jest.fn(() => ({ remove: jest.fn() })),
  removeListener: jest.fn(),
  removeAllListeners: jest.fn(),
}));

module.exports = {
  // Components
  View,
  Text,
  TouchableOpacity,
  Image,
  TextInput,
  ScrollView,
  FlatList,
  Modal,
  ActivityIndicator,
  SafeAreaView,
  KeyboardAvoidingView,
  Pressable,
  Button,
  Switch,
  RefreshControl,
  
  // APIs
  Platform,
  StyleSheet,
  Dimensions,
  Alert,
  Animated,
  Keyboard,
  PixelRatio,
  NativeModules,
  NativeEventEmitter,
  
  // Utilities
  LayoutAnimation: {
    configureNext: jest.fn(),
    create: jest.fn(),
    Types: {},
    Properties: {},
  },
  
  StatusBar: {
    setBarStyle: jest.fn(),
    setBackgroundColor: jest.fn(),
    setHidden: jest.fn(),
  },
  
  AppState: {
    currentState: 'active',
    addEventListener: jest.fn(() => ({ remove: jest.fn() })),
    removeEventListener: jest.fn(),
  },
  
  Linking: {
    openURL: jest.fn(() => Promise.resolve()),
    canOpenURL: jest.fn(() => Promise.resolve(true)),
    addEventListener: jest.fn(() => ({ remove: jest.fn() })),
    removeEventListener: jest.fn(),
    getInitialURL: jest.fn(() => Promise.resolve(null)),
  },
};
