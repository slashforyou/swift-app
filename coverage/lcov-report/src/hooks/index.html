
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/hooks</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/hooks</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">11.17% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>163/1459</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">7.66% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>46/600</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">14.87% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>36/242</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">11.02% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>150/1360</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="useCalendar-fixed.ts"><a href="useCalendar-fixed.ts.html">useCalendar-fixed.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="36" class="abs low">0/36</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="17" class="abs low">0/17</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	</tr>

<tr>
	<td class="file low" data-value="useCalendar.ts"><a href="useCalendar.ts.html">useCalendar.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="37" class="abs low">0/37</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="19" class="abs low">0/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="33" class="abs low">0/33</td>
	</tr>

<tr>
	<td class="file low" data-value="useClients.ts"><a href="useClients.ts.html">useClients.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="33" class="abs low">0/33</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="32" class="abs low">0/32</td>
	</tr>

<tr>
	<td class="file low" data-value="useCommonStyles.ts"><a href="useCommonStyles.ts.html">useCommonStyles.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	</tr>

<tr>
	<td class="file low" data-value="useGamification.ts"><a href="useGamification.ts.html">useGamification.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="64" class="abs low">0/64</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="58" class="abs low">0/58</td>
	</tr>

<tr>
	<td class="file low" data-value="useGamificationFixed.ts"><a href="useGamificationFixed.ts.html">useGamificationFixed.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="49" class="abs low">0/49</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="43" class="abs low">0/43</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobDetails.ts"><a href="useJobDetails.ts.html">useJobDetails.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="133" class="abs low">0/133</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="52" class="abs low">0/52</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="127" class="abs low">0/127</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobNotes.ts"><a href="useJobNotes.ts.html">useJobNotes.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="99" class="abs low">0/99</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="40" class="abs low">0/40</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="15" class="abs low">0/15</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="92" class="abs low">0/92</td>
	</tr>

<tr>
	<td class="file medium" data-value="useJobPhotos.ts"><a href="useJobPhotos.ts.html">useJobPhotos.ts</a></td>
	<td data-value="51.75" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 51%"></div><div class="cover-empty" style="width: 49%"></div></div>
	</td>
	<td data-value="51.75" class="pct medium">51.75%</td>
	<td data-value="114" class="abs medium">59/114</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="42" class="abs low">15/42</td>
	<td data-value="38.88" class="pct low">38.88%</td>
	<td data-value="18" class="abs low">7/18</td>
	<td data-value="54.71" class="pct medium">54.71%</td>
	<td data-value="106" class="abs medium">58/106</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobPhotos_temp.ts"><a href="useJobPhotos_temp.ts.html">useJobPhotos_temp.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="114" class="abs low">0/114</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="42" class="abs low">0/42</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="106" class="abs low">0/106</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobTimer.ts"><a href="useJobTimer.ts.html">useJobTimer.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="121" class="abs low">0/121</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="83" class="abs low">0/83</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="115" class="abs low">0/115</td>
	</tr>

<tr>
	<td class="file high" data-value="useJobsBilling.ts"><a href="useJobsBilling.ts.html">useJobsBilling.ts</a></td>
	<td data-value="94.36" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 94%"></div><div class="cover-empty" style="width: 6%"></div></div>
	</td>
	<td data-value="94.36" class="pct high">94.36%</td>
	<td data-value="71" class="abs high">67/71</td>
	<td data-value="63.04" class="pct medium">63.04%</td>
	<td data-value="46" class="abs medium">29/46</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="93.75" class="pct high">93.75%</td>
	<td data-value="64" class="abs high">60/64</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobsForDay.ts"><a href="useJobsForDay.ts.html">useJobsForDay.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="109" class="abs low">0/109</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="77" class="abs low">0/77</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="104" class="abs low">0/104</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobsForMonth.ts"><a href="useJobsForMonth.ts.html">useJobsForMonth.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="52" class="abs low">0/52</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="51" class="abs low">0/51</td>
	</tr>

<tr>
	<td class="file low" data-value="useJobsForYear.ts"><a href="useJobsForYear.ts.html">useJobsForYear.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="44" class="abs low">0/44</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="13" class="abs low">0/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="44" class="abs low">0/44</td>
	</tr>

<tr>
	<td class="file low" data-value="useSession-fixed.ts"><a href="useSession-fixed.ts.html">useSession-fixed.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="22" class="abs low">0/22</td>
	</tr>

<tr>
	<td class="file low" data-value="useSession.ts"><a href="useSession.ts.html">useSession.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="56" class="abs low">0/56</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="55" class="abs low">0/55</td>
	</tr>

<tr>
	<td class="file low" data-value="useStaff.ts"><a href="useStaff.ts.html">useStaff.ts</a></td>
	<td data-value="43.02" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 43%"></div><div class="cover-empty" style="width: 57%"></div></div>
	</td>
	<td data-value="43.02" class="pct low">43.02%</td>
	<td data-value="86" class="abs low">37/86</td>
	<td data-value="14.28" class="pct low">14.28%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="24" class="abs low">10/24</td>
	<td data-value="43.83" class="pct low">43.83%</td>
	<td data-value="73" class="abs low">32/73</td>
	</tr>

<tr>
	<td class="file low" data-value="useUserProfile.ts"><a href="useUserProfile.ts.html">useUserProfile.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="94" class="abs low">0/94</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="92" class="abs low">0/92</td>
	</tr>

<tr>
	<td class="file low" data-value="useVehicles.ts"><a href="useVehicles.ts.html">useVehicles.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="120" class="abs low">0/120</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="30" class="abs low">0/30</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="107" class="abs low">0/107</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2025-10-26T07:39:51.817Z
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    