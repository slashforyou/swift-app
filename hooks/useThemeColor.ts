/**
 * Enhanced theme color hook with comprehensive color support
 * Learn more about light and dark modes:
 * https://docs.expo.dev/guides/color-schemes/
 */

import { Colors, ColorName, ColorScheme, getColorWithOpacity } from '@/src/constants/Colors';
import { useColorScheme } from '@/hooks/useColorScheme';
import { useTheme, useThemeColors as useColors } from '../src/context/ThemeProvider';

/**
 * Get a specific theme color (legacy support)
 */
export function useThemeColor(
  props: { light?: string; dark?: string },
  colorName: ColorName
) {
  try {
    // Try to use the new ThemeProvider context first
    const { theme, colors } = useTheme();
    const colorFromProps = props[theme];

    if (colorFromProps) {
      return colorFromProps;
    } else {
      return colors[colorName];
    }
  } catch {
    // Fallback to the old system if ThemeProvider is not available
    const theme = useColorScheme() ?? 'light';
    const colorFromProps = props[theme];

    if (colorFromProps) {
      return colorFromProps;
    } else {
      return Colors[theme][colorName];
    }
  }
}

/**
 * Get all theme colors for current scheme
 */
export function useThemeColors() {
  try {
    return useColors();
  } catch {
    // Fallback to the old system
    const theme = useColorScheme() ?? 'light';
    return Colors[theme];
  }
}

/**
 * Get current color scheme
 */
export function useCurrentTheme(): ColorScheme {
  try {
    const { theme } = useTheme();
    return theme;
  } catch {
    return useColorScheme() ?? 'light';
  }
}

/**
 * Get a color with opacity
 */
export function useColorWithOpacity(colorName: ColorName, opacity: number) {
  const colors = useThemeColors();
  return getColorWithOpacity(colors[colorName], opacity);
}

/**
 * Hook for getting multiple themed colors at once
 */
export function useThemedColors<T extends Record<string, ColorName>>(
  colorMap: T
): Record<keyof T, string> {
  const colors = useThemeColors();
  
  const result = {} as Record<keyof T, string>;
  
  for (const [key, colorName] of Object.entries(colorMap)) {
    result[key as keyof T] = colors[colorName];
  }
  
  return result;
}

/**
 * Create themed styles hook
 */
export function useThemedStyles<T>(
  stylesFn: (colors: typeof Colors.light) => T
): T {
  const colors = useThemeColors();
  return stylesFn(colors);
}
