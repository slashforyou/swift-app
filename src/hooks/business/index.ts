/**
 * Index des hooks business
 * Hooks React pour la gestion compl√®te de l'entreprise
 */

// Hooks business principaux
export { useBusinessInfo } from './useBusinessInfo';
export { useBusinessStaff } from './useBusinessStaff';
export { useBusinessVehicles } from './useBusinessVehicles';
export { useInvoices } from './useInvoices';
export { useJobTemplates } from './useJobTemplates';

// Hook composite
export { useBusinessManager } from './useBusinessManager';

// Types des hooks (pour faciliter l'usage)
export type {
    BusinessInfo, BusinessStaff, BusinessStats,
    BusinessVehicle, Invoice, InvoiceCreateData, InvoiceItem, JobTemplate, StaffCreateData, TemplateCreateData, VehicleCreateData
} from '../../services/business';
