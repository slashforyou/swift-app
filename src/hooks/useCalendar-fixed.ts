import { useState, useEffect } from 'react'
import loadCalendarDays, { CalendarDay, CalendarApiError } from '../services/calendar/loadCalendarDays'

interface UseCalendarState {
  data: CalendarDay[]
  isLoading: boolean
  error: string | null
  lastFetch: Date | null
}

interface UseCalendarReturn extends UseCalendarState {
  loadDays: (startDate: Date, endDate: Date) => Promise<void>
  refresh: () => Promise<void>
  clearError: () => void
  hasData: boolean
}

/**
 * Custom hook for managing calendar data - Version simplifiée
 */
export function useCalendar(): UseCalendarReturn {
  const [state, setState] = useState<UseCalendarState>({
    data: [],
    isLoading: false,
    error: null,
    lastFetch: null
  })

  const [lastQuery, setLastQuery] = useState<{
    startDate: Date
    endDate: Date
  } | null>(null)

  const clearError = () => {
    setState(prev => ({ ...prev, error: null }))
  }

  const loadDays = async (startDate: Date, endDate: Date) => {
    try {
      setState(prev => ({ ...prev, isLoading: true, error: null }))
      
      const data = await loadCalendarDays(startDate, endDate)
      
      setState(prev => ({
        ...prev,
        data,
        isLoading: false,
        lastFetch: new Date()
      }))
      
      setLastQuery({ startDate, endDate })
      
    } catch (error) {
      let errorMessage = 'Failed to load calendar data'
      
      if (error instanceof CalendarApiError) {
        errorMessage = error.message
      } else if (error instanceof Error) {
        errorMessage = error.message
      }
      
      setState(prev => ({
        ...prev,
        isLoading: false,
        error: errorMessage
      }))
      
      if (__DEV__) {
        console.error('Calendar loading error:', error)
      }
    }
  }

  const refresh = async () => {
    if (lastQuery) {
      await loadDays(lastQuery.startDate, lastQuery.endDate)
    }
  }

  return {
    ...state,
    loadDays,
    refresh,
    clearError,
    hasData: state.data.length > 0
  }
}

/**
 * Hook for loading calendar data with initial load
 * Version simplifiée sans boucles infinies
 */
export function useCalendarData(
  startDate: Date,
  endDate: Date,
  autoLoad: boolean = true
) {
  const calendar = useCalendar()
  const [hasInitialLoaded, setHasInitialLoaded] = useState(false)

  // Chargement initial une seule fois
  useEffect(() => {
    if (autoLoad && !hasInitialLoaded) {
      calendar.loadDays(startDate, endDate)
      setHasInitialLoaded(true)
    }
  }, []) // Pas de dépendances = une seule fois

  // Recharger seulement si les dates changent ET qu'on a déjà fait un chargement initial
  useEffect(() => {
    if (hasInitialLoaded && autoLoad) {
      calendar.loadDays(startDate, endDate)
    }
  }, [startDate.getTime(), endDate.getTime()]) // Utiliser getTime() pour éviter les références d'objets

  return calendar
}