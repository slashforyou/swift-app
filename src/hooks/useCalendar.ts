import React, { useState, useCallback, useEffect, useRef } from 'react'
import loadCalendarDays, { CalendarDay, CalendarApiError } from '../services/calendar/loadCalendarDays'

interface UseCalendarState {
  data: CalendarDay[]
  isLoading: boolean
  error: string | null
  lastFetch: Date | null
}

interface UseCalendarReturn extends UseCalendarState {
  loadDays: (startDate: Date, endDate: Date) => Promise<void>
  refresh: () => Promise<void>
  clearError: () => void
  hasData: boolean
}

/**
 * Custom hook for managing calendar data
 * Provides loading state, error handling, and caching
 */
export function useCalendar(): UseCalendarReturn {
  const [state, setState] = useState<UseCalendarState>({
    data: [],
    isLoading: false,
    error: null,
    lastFetch: null
  })

  const [lastQuery, setLastQuery] = useState<{
    startDate: Date
    endDate: Date
  } | null>(null)

  const clearError = useCallback(() => {
    setState(prev => ({ ...prev, error: null }))
  }, [])

  const loadDays = async (startDate: Date, endDate: Date) => {

    if (__DEV__) {
      console.log("***** LoadDays called *****");
    }
    try {
      setState(prev => ({ ...prev, isLoading: true, error: null }))
      
      const data = await loadCalendarDays(startDate, endDate)
      
      setState(prev => ({
        ...prev,
        data,
        isLoading: false,
        lastFetch: new Date()
      }))
      
      setLastQuery({ startDate, endDate })
      
    } catch (error) {
      let errorMessage = 'Failed to load calendar data'
      
      if (error instanceof CalendarApiError) {
        errorMessage = error.message
      } else if (error instanceof Error) {
        errorMessage = error.message
      }
      
      setState(prev => ({
        ...prev,
        isLoading: false,
        error: errorMessage
      }))
      
      if (__DEV__) {
        console.error('Calendar loading error:', error)
      }
    }
  }

  const refresh = async () => {
    if (lastQuery) {
      await loadDays(lastQuery.startDate, lastQuery.endDate)
    }
  }

  return {
    ...state,
    loadDays,
    refresh,
    clearError,
    hasData: state.data.length > 0
  }
}

/**
 * Hook for loading calendar data with automatic retry and caching
 * @param startDate Initial start date
 * @param endDate Initial end date
 * @param autoLoad Whether to load data immediately
 */
export function useCalendarData(
  startDate: Date,
  endDate: Date,
  autoLoad: boolean = true
) {
  const calendar = useCalendar()
  const hasLoaded = useRef(false)

    //   If autoLoad is true, load data on mount
    // Console log for debug if the user is authenticated
    if (autoLoad && !hasLoaded.current) {
      console.log('User is authenticated, loading calendar data...')
    }

  // Auto-load data seulement une fois au mount
  useEffect(() => {
    if (autoLoad && !hasLoaded.current) {
      hasLoaded.current = true
      calendar.loadDays(startDate, endDate)
    }
  }, []) // Pas de d√©pendances = une seule fois

  return calendar
}