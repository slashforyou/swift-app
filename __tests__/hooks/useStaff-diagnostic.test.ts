/**
 * Test de diagnostic pour useStaff
 */
import { act, renderHook, waitFor } from '@testing-library/react-native';

// Démock useStaff pour tester le vrai hook
jest.unmock('../../src/hooks/useStaff');

import { useStaff } from '../../src/hooks/useStaff';

describe('useStaff Diagnostic', () => {
  it('should load data after refreshStaff', async () => {
    const { result } = renderHook(() => useStaff());
    
    const beforeKeys = Object.keys(result.current);
    const beforeStaffLength = result.current.staff.length;
    
    // Essayons de charger
    await act(async () => {
      await result.current.refreshStaff();
    });
    
    const afterStaffLength = result.current.staff.length;
    const afterEmployeesLength = result.current.employees.length;
    const afterContractorsLength = result.current.contractors.length;
    
    // Snapshot pour voir ce qui se passe
    expect({
      beforeKeys,
      beforeStaffLength,
      afterStaffLength,
      afterEmployeesLength,
      afterContractorsLength,
      hasRefreshStaff: typeof result.current.refreshStaff,
      hasRefreshData: typeof result.current.refreshData,
    }).toMatchSnapshot();
    
    // Test réel
    expect(afterStaffLength).toBeGreaterThan(0);
  });
});
