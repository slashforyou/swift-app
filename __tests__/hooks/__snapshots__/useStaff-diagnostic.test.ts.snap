// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`useStaff Diagnostic should load data after refreshStaff 1`] = `
{
  "afterContractorsLength": 0,
  "afterEmployeesLength": 0,
  "afterStaffLength": 0,
  "beforeKeys": [
    "staff",
    "employees",
    "contractors",
    "inviteEmployee",
    "searchContractor",
    "addContractor",
    "updateEmployee",
    "updateContractor",
    "deleteEmployee",
    "deleteContractor",
    "refreshStaff",
    "isLoading",
    "error",
    "totalActive",
    "totalEmployees",
    "totalContractors",
    "totalTeams",
    "averageEmployeeRate",
  ],
  "beforeStaffLength": 0,
  "hasRefreshData": "undefined",
  "hasRefreshStaff": "function",
}
`;
